src/sources/aws_s3/sqs.rs

```diff
@@ -293,6 +282,34 @@ impl IngestorProcess {
                 }
             }
         }
+
+        if !delete_entries.is_empty() {
+            // We need these for a correct error message if the batch fails overall.
+            let cloned_entries = delete_entries.clone();
+            match self.delete_messages(delete_entries).await {
```

Comment:
Too bad you can't pass this by reference. This clone is otherwise pretty pointless (since the internal events can be written to use a reference and not need a clone).

Reply:
Yeah, it's annoying. :(